pkg/check/datastore.go

```diff
@@ -329,33 +341,33 @@ func checkDataStoreWithURL(ctx *CheckContext, dsURL string, dsTypes dataStoreTyp
 	klog.V(2).Infof("checking datastore %s for permissions", dsURL)
 	var errs []error
 
-	dsMo, dcName, err := getDatastoreByURL(ctx, dsURL)
+	dsMo, dcName, vCenter, err := getDatastoreByURL(ctx, dsURL)
 	if err != nil {
 		return err
 	}
 
-	if err := checkForDatastoreCluster(ctx, dsMo, dsURL, dcName, dsTypes); err != nil {
+	if err := checkForDatastoreCluster(ctx, vCenter, dsMo, dsURL, dcName, dsTypes); err != nil {
 		errs = append(errs, err)
 	}
-	if err := checkDatastorePrivileges(ctx, dsURL, dsMo.Reference()); err != nil {
+	if err := checkDatastorePrivileges(ctx, vCenter, dsURL, dsMo.Reference()); err != nil {
 		errs = append(errs, err)
 	}
 	return errors.NewAggregate(errs)
 }
 
-func checkForDatastoreCluster(ctx *CheckContext, dsMo mo.Datastore, dataStoreName, dcName string, dsTypes dataStoreTypeCollector) error {
+func checkForDatastoreCluster(ctx *CheckContext, vCenter *VCenter, dsMo mo.Datastore, dataStoreName, dcName string, dsTypes dataStoreTypeCollector) error {
 	// Collect DS type
 	dsType := dsMo.Summary.Type
 	klog.V(4).Infof("Datastore %s is of type %s", dataStoreName, dsType)
 	dsTypes.addDataStore(dataStoreName, dsType)
 
-	content, err := ctx.Cache.GetStoragePods(ctx.Context)
+	content, err := vCenter.Cache.GetStoragePods(ctx.Context)
 	if err != nil {
 		return err
```

Comment:
Not sure I am following this request.  The cache was just moved into the VCenter object instead of root of ctx.  Are you wanting me to change the `return err` to now have a wrapper around it?